### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by balanalina**

### CODE REVIEW

File: app/assets/stylesheets/cart_items.scss [hunk 1]
```diff
@@ -0,0 +1,37 @@
+// Place all the styles related to the CartItems controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
+.cart-picture {
+  height: 250px;
+  display: flex;
+  padding-top: 7px;
+
+  img{
```
Comment (florinionce) [inline app/assets/stylesheets/cart_items.scss:9]: you should add spaces before curly braces :D

File: app/controllers/cart_items_controller.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
+class CartItemsController < ApplicationController
+  include CartsHelper
+  before_action :set_cart_item, only: %i[show edit update destroy]
+  before_action :set_current_cart, only: :create
+
+  def index
+    @cart_items = CartItem.all
```
Comment (florinionce) [inline app/controllers/cart_items_controller.rb:7]: how is this working when you have multiple users adding products to their carts?

File: app/controllers/cart_items_controller.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  end
+
+  def edit; end
+
+  def create
+    product = Product.find(params[:product_id])
+    @cart_item = @cart.add_product(product)
+    respond_to do |format|
+      if @cart_item.save
+        format.html { redirect_to root_path, notice: 'Item successfully added to the cart!' }
```
Comment (florinionce) [inline app/controllers/cart_items_controller.rb:23]: why do you need `format.html`? :)

File: app/controllers/cart_items_controller.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      else
+        format.html { redirect_to root_path, notice: 'Could not add the item to cart!' }
+      end
+    end
+  end
+
+  def destroy
+    @cart = current_user.cart
```
Comment (florinionce) [inline app/controllers/cart_items_controller.rb:31]: why do you set the cart from the `set_current_cart` helper method for the `#create` method and you use it from the `current_user` here?

File: app/controllers/carts_controller.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  before_action :set_cart, only: %i[show edit destroy]
+
+  def show; end
+
+  def destroy
+    @cart.cart_items.delete_all
+    redirect_to(@cart)
+  end
+
+  def edit; end
```
Comment (florinionce) [inline app/controllers/carts_controller.rb:12]: is this ever used?

File: app/controllers/carts_controller.rb [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  private
+
+
+  def set_cart
+    @cart = current_user.cart
+  end
+
+  def invalid_cart
+    logger.error "Attempt to access invalid carts #{params[:user_id]}"
+    redirect_to root_path, notice: "That carts doesn't exist"
```
Comment (florinionce) [inline app/controllers/carts_controller.rb:23]: shouldn't you also create the cart for that user? :D Just a suggestion

File: app/models/cart_item.rb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+class CartItem < ApplicationRecord
+  belongs_to :product
+  belongs_to :cart
+
+  def total_price
+    product.price.to_i * quantity.to_i
```
Comment (florinionce) [inline app/models/cart_item.rb:6]: is the price stored as integer or as float?

File: app/views/cart_items/_cart_item.haml [hunk 8]
```diff
@@ -0,0 +1,14 @@
+.columns.align-items-center
+  .column.is-1
+    .cart-picture
+      =image_tag(url_for(cart_item.product.image))
```
Comment (florinionce) [inline app/views/cart_items/_cart_item.haml:4]: I think you should leave one empty space after the equals sign

File: app/views/layouts/_dishes.haml [hunk 9]
```diff
@@ -9,4 +7,9 @@
         %b
           %p.title= dish.title
           %p.category= dish.category
-          %p.price= '$'+dish.price.round.to_s
+          %p.price= number_to_currency(dish.price)
+          - if logged_in?
+            =button_to 'Add to cart',cart_items_path(product_id: dish.id) ,class: 'button carts-button is-warning add-to-carts'
+          - else
+            =button_to 'Add to cart', new_user_url , method: :get ,class: 'button carts-button is-warning add-to-carts'
+
```
Comment (florinionce) [inline app/views/layouts/_dishes.haml:15]: extra line here :D

File: config/local_env.yml [hunk 10]
```diff
@@ -0,0 +1,3 @@
+CLOUD_NAME: dwdawkjo5
```
Comment (florinionce) [inline config/local_env.yml:1]: isn't this supposed to be gitignored?

File: db/seeds.rb [hunk 11]
```diff
@@ -5,3 +5,29 @@
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
+#
+
+new_product = Product.create_or_find_by!(title: 'Eggs', category: 'Breakfast', price: 12)
```
Comment (florinionce) [inline db/seeds.rb:10]: you could dry all this code if you defined your data in an array of hashes and then iterate through it.

File: spec/models/user_spec.rb [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+      expect(user2.valid?).to eq false
+    end
+
+    it 'too short email' do
+      @user.email= 'a@a.com'
+      expect(@user.valid?).to eq false
+      @user.email= 'examples@valid.com'
+    end
+
+
```
Comment (florinionce) [inline spec/models/user_spec.rb:44]: extra lines

File: config/local_env.yml (hunk 10 above)
Comment (balanalina) [reply config/local_env.yml:1]: It should be gitignored :-? 
image

File: app/controllers/cart_items_controller.rb (hunk 4 above)
Comment (balanalina) [reply app/controllers/cart_items_controller.rb:31]: I set the helper method for #destroy too, I don't know why I did it like that

File: app/controllers/cart_items_controller.rb (hunk 2 above)
Comment (balanalina) [reply app/controllers/cart_items_controller.rb:7]: I don't use it yet, I will just delete it

**Pull request merged**